### PR TITLE
rails: enforce unique entity names

### PR DESCRIPTION
**Pull request by aspiers**
This should prevent having multiple records for people, organisations, etc. with the same name.

I am not 100% sure if this is correct; would appreciate reviews especially from @JohnSmall and @Greatlemer ;-)

### CODE REVIEW

**Comment by Greatlemer** [comment]
@aspiers, The problem with this is that there could very well be people with the same name, for them the uniqueness may need to be over a combination of their role and their name (although in theory that may not be unique either)

**Comment by aspiers** [comment]
Yeah, that's true with people. Maybe it should only apply for organisations and government offices then?

**Comment by Greatlemer** [comment]
Yeah, I think that may be the best we can go for at the moment.

**Review by elischutze — COMMENTED** [review]
do we still want to update this to be orgs and offices only?

**Comment by aspiers** [comment]
Yeah I think we do.  I'll move the validation to the subclasses and rebase.

**Comment by aspiers** [comment]
Done.

File: src/ruby/meetings/spec/models/entity_spec.rb [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
-RSpec.describe GovernmentOffice,type: :model do
-  it{ should have_many(:influence_government_office_people)}
-  it{ should have_many(:meetings).through(:influence_government_office_people)}
-  it{ should have_many(:hospitalities).through(:influence_government_office_people)}
-  it{ should have_many(:gifts).through(:influence_government_office_people)}
-  it{ should have_many(:travels).through(:influence_government_office_people)}
+RSpec.describe GovernmentOffice, type: :model do
+  it { should have_many(:influence_government_office_people) }
+  it { should have_many(:meetings).through(:influence_government_office_people) }
+  it { should have_many(:hospitalities).through(:influence_government_office_people) }
```
[the remark this answers is not in the document]
Comment (aspiers) [reply src/ruby/meetings/spec/models/entity_spec.rb:31]: I think in this context, leaving the long line is an acceptable compromise - it's better than the alternatives.

**Review by Greatlemer — APPROVED** [review]

**Pull request merged**